src/Views/PartitioningView.vala

```diff
@@ -0,0 +1,302 @@
+        label_sizer = new Gtk.SizeGroup (Gtk.SizeGroupMode.BOTH);
+
+        foreach (unowned Distinst.Disk disk in disks.list ()) {
+            // Skip root disk or live disk
+            if (disk.contains_mount ("/") || disk.contains_mount ("/cdrom")) {
+                continue;
+            }
+
+            var sector_size = disk.get_sector_size ();
+            var size = disk.get_sectors () * sector_size;
+
+            string path = Utils.string_from_utf8 (disk.get_device_path ());
+
+            string model = disk.get_model ();
+            string label = (model.length == 0)
+                ? disk.get_serial ().replace ("_", " ")
+                : model;
+
+            var partitions = new Gee.ArrayList<PartitionBar> ();
+            foreach (unowned Distinst.Partition part in disk.list_partitions ()) {
+                var partition = new PartitionBar (part, path, sector_size, false, this.set_mount, this.unset_mount, this.mount_is_set, this.decrypt);
+                partitions.add (partition);
+            }
+
+            var disk_bar = new DiskBar (model, path, size, (owned) partitions);
+            label_sizer.add_widget (disk_bar.label);
+            disk_list.pack_start (disk_bar);
+        }
+
+        foreach (unowned Distinst.LvmDevice disk in disks.list_logical ()) {
+            add_logical_disk (disk);
+        }
+
+        disk_list.show_all ();
+    }
+
+    private void open_gparted () {
+        try {
+            var process = new GLib.Subprocess.newv ({"gparted"}, GLib.SubprocessFlags.NONE);
+            process.wait ();
+        } catch (GLib.Error error) {
+            stderr.printf ("critical error occurred when executing gparted\n");
+        }
+
+        reset_view ();
+    }
+
+    private void reset_view () {
+        disk_list.get_children ().foreach ((child) => child.destroy ());
+        mounts.clear ();
+        luks.clear ();
+        next_button.sensitive = false;
+        load_disks ();
+    }
+
+    private void add_logical_disk (Distinst.LvmDevice disk) {
+        var sector_size = disk.get_sector_size ();
+        var size = disk.get_sectors () * sector_size;
+
+        string path = Utils.string_from_utf8 (disk.get_device_path ());
+
+        string model = disk.get_model ();
+
+        var partitions = new Gee.ArrayList<PartitionBar> ();
+        foreach (unowned Distinst.Partition part in disk.list_partitions ()) {
+            var partition = new PartitionBar (part, path, sector_size, true, this.set_mount, this.unset_mount, this.mount_is_set, this.decrypt);
+            partitions.add (partition);
+        }
+
+        var disk_bar = new DiskBar (model, path, size, (owned) partitions);
+        label_sizer.add_widget (disk_bar.label);
+        disk_list.pack_start (disk_bar);
+    }
+
+    private void validate_status () {
+        uint8 flags = 0;
+        const uint8 ROOT = 1;
+        const uint8 BOOT = 2;
+
+        stderr.printf ("DEBUG: Current Layout:\n");
```

Comment:
please store everything in a string and use the `debug` function which allows a fine-grained level of debug outputs